lib/exercism/authentication.rb

```diff
@@ -1,6 +1,36 @@
 class Authentication
   def self.perform(code, client_id, client_secret, target_profile)
-    id, username, email, avatar_url = Github.authenticate(code, client_id, client_secret)
-    User.from_github(id, username, email, avatar_url, target_profile)
+    new(code, client_id, client_secret, target_profile).authenticate
+  end
+
+  def initialize(code, client_id, client_secret, target_profile)
```

Comment:
I'm not sure what you're proposing to replace.

There are a couple of things to consider here.

1. This proposal means `Authentication` needs to know too much about the guts of how `GitHubAuth` works.
2. User needs to know (too much?) about how github authentication works.
3. There is only one Authentication method.

What we have: `code, client_id, client_secret, target_profile`
What we need: instance of an Exercism `User`

What is the ideal code you would write to make that happen?